vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
The real issue here is I probably shouldn't still be looking at my computer screen, and I have made this more confusing throughout by misunderstanding some of your comments. 

I think we have established that I intended the code to be the way it is in this PR.

So a comment revision:
```cpp     
 // if the RequestOrResponse (result) has been populated with success=false 
//  and an error message then we have enough information to proceed with a conversion, 
//  writing out an error response
//  if err is empty, we have a successful RequestOrResponse so again proceed with conversion
//  however if it isn't empty, something went wrong whilst parsing, so throw
```